src/parser/semantic_check.c

```diff
@@ -2298,6 +2298,10 @@ pt_is_compatible_without_cast (PARSER_CONTEXT * parser, SEMAN_COMPATIBLE_INFO *
 
   if (dest_sci->type_enum != src->type_enum)
     {
+      if (src->type_enum == PT_TYPE_MAYBE && dest_sci->type_enum == PT_TYPE_NUMERIC)
```

Comment:
If it is called through pt_values_query_to_compatible_cast() or pt_to_compatible_cast(), can it never happen if the following conditions are true?
(src->type_enum == PT_TYPE_MAYBE && dest_sci->type_enum == PT_TYPE_NUMERIC)

Reply:
I think, in general, PT_TYPE_MAYBE to PT_TYPE_NUMERIC is compatible without cast. In any similar case of this, the cast is not needed at compile time. It will be automatically done by the coercing instead of casting at execution time.

Reply:
The code is reverted and I modified the function pt_assignment_compatible as @hgryoo suggested.